lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -174,6 +174,74 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     Enum.reverse(withdrawals)
   end
 
+  @spec process_proposer_slashing(BeaconState.t(), SszTypes.ProposerSlashing.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_proposer_slashing(state, proposer_slashing) do
+    header_1 = proposer_slashing.signed_header_1.message
+    header_2 = proposer_slashing.signed_header_2.message
+    proposer = Enum.at(state.validators, header_1.proposer_index)
+
+    cond do
+      not (header_1.slot == header_2.slot) ->
+        {:error, "Slots doesn't match"}
+
+      not (header_1.proposer_index == header_2.proposer_index) ->
+        {:error, "Proposer indices doesn't match"}
+
+      not (header_1 != header_2) ->
+        {:error, "Headers are same"}
+
+      not Predicates.is_slashable_validator(proposer, Accessors.get_current_epoch(state)) ->
+        {:error, "Proposer is not slashable"}
+
+      true ->
+        is_verified =
+          [proposer_slashing.signed_header_1, proposer_slashing.signed_header_2]
+          |> Enum.reduce_while(false, fn signed_header, _acc ->
```

Comment:
I suggest using [`all?`](https://hexdocs.pm/elixir/Enum.html#all?/2) for this, like so:

```elixir
|> Enum.all?(fn signed_header -> ... end)
```